### PR TITLE
adds functionality to drop input columns in CombineWithReferenceFeature and MatchCombination

### PR DESCRIPTION
**Pull request by kylegilde**
adding the drop_original argument to MathematicalCombination and CombineWithReferenceFeature

### CODE REVIEW

File: feature_engine/creation/combine_with_reference_feature.py [hunk 1]
```diff
@@ -65,6 +65,8 @@ class CombineWithReferenceFeature(BaseEstimator, TransformerMixin):
         transformer will return an error if the training or the datasets to transform
         contain missing values.
 
+    drop_original: bool, default=False
+        If True, the original variables to transform will be dropped from the dataframe.
```
Comment (solegalli) [inline feature_engine/creation/combine_with_reference_feature.py:69]: I would change the wording to:

If True, the original variables will be dropped from the dataframe after their combination.

File: feature_engine/creation/mathematical_combination.py [hunk 2]
```diff
@@ -61,6 +61,9 @@ class MathematicalCombination(BaseEstimator, TransformerMixin):
         contain missing values. If 'ignore', missing data will be ignored when
         performing the calculations.
 
+    drop_original: bool, default=False
+        If True, the original variables to transform will be dropped from the dataframe.
```
Comment (solegalli) [inline feature_engine/creation/mathematical_combination.py:65]: I would change the wording to:

If True, the original variables will be dropped from the dataframe after their combination.

File: feature_engine/creation/combine_with_reference_feature.py [hunk 3]
```diff
@@ -286,6 +293,9 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         if self.new_variables_names:
             X.columns = original_col_names + self.new_variables_names
 
+        if self.drop_original:
+            X.drop(columns=self.variables_, inplace=True)
```
Comment (solegalli) [inline feature_engine/creation/combine_with_reference_feature.py:297]: here we need to drop `self.variables_co_combine + self.reference_variables`.

I don't think this transformer has the attribute `variables_`, does it?

File: feature_engine/creation/mathematical_combination.py [hunk 4]
```diff
@@ -251,6 +259,9 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         for new_variable_name, operation in self.combination_dict_.items():
             X[new_variable_name] = X[self.variables_to_combine].agg(operation, axis=1)
 
+        if self.drop_original:
+            X.drop(columns=self.variables_, inplace=True)
```
Comment (solegalli) [inline feature_engine/creation/mathematical_combination.py:263]: here we need to drop `self.variables_to_combine`.

**Review by solegalli — COMMENTED** [review]
Hi @kylegilde 

thanks a lot for the addition to the functionality. 

I made a few comments regarding wording of the docs, and the variables that need to be dropped from the dataframe. If I understood correctly, we will be dropping the variables that we combine mathematically, correct?

Also, we would need to add a couple of tests for each of the transformers to confirm that when drop_original=True, the original variables are dropped.

Would you mind adding those tests please?


PS: maybe you were going to add those already and I jumped ahead of my game, if so, apologies.

File: feature_engine/creation/combine_with_reference_feature.py (hunk 1 above)
Comment (kylegilde) [reply feature_engine/creation/combine_with_reference_feature.py:69]: Oh, I just copied and pasted the wording from CyclicalTransformer w/o even reading it. Should we update that wording too?

File: feature_engine/creation/mathematical_combination.py (hunk 4 above)
Comment (kylegilde) [reply feature_engine/creation/mathematical_combination.py:263]: > 

Can you provide an example of how to write a test? Thank you

File: feature_engine/creation/combine_with_reference_feature.py (hunk 1 above)
Comment (solegalli) [reply feature_engine/creation/combine_with_reference_feature.py:69]: Just here please :)

File: feature_engine/creation/mathematical_combination.py (hunk 4 above)
Comment (solegalli) [reply feature_engine/creation/mathematical_combination.py:263]: In this [code block](https://github.com/feature-engine/feature_engine/blob/main/tests/test_creation/test_cyclical_transformer.py#L75), we extend the first test of the cyclical transformer to test the drop_original attribute.

In short, I would suggest you pick one of the tests that already exist for each of the transformers, and replicate it, changing the expected output when drop_original=True

**Comment by solegalli** [comment]
@dodoarg if you have time, maybe you would like to pick this one up? :p 

So we can squeeze it in the next release.

**Pull request closed without merging**